src/backend/internal/controller/user/reg_customer.go

```diff
@@ -0,0 +1,50 @@
+package user
+
+import (
+	"net/http"
+
+	"github.com/Roongkun/software-eng-ii/internal/controller/user/fieldvalidate"
+	"github.com/Roongkun/software-eng-ii/internal/controller/util"
+	"github.com/Roongkun/software-eng-ii/internal/model"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+)
+
+func (r *Resolver) RegCustomer(c *gin.Context) {
+	newUser := model.UserInput{}
+	if err := c.BindJSON(&newUser); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
```

Comment:
consider adding `"status": "failed"` to the returning JSON.

In fact, the `status` field should be added to every returning JSON, with `"status": "success"` indicating successful requests, and `"status": "failed"` indicating errors. 

This way, when the front-end receive the response, they could simply abort operation if `"status==failed"`, and continue if otherwise.